chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -383,9 +383,22 @@ def unique_name
         end
 
         def claimed?
-          not @node[:crowbar_wall][:claimed_disks][self.unique_name][:owner].to_s.empty?
-        rescue
-          false
+          claimed_disks = @node[:crowbar_wall][:claimed_disks].to_hash
+          claimed_disks.each do | claimed_name, v |
+            claimed_real = File.realpath(claimed_name)
+            if claimed_real == self.name and not v[:owner].empty?
+              Chef::Log.info("Disk #{self.unique_name} (#{self.name}) already " \
+                             "claimed by #{claimed_name} (#{claimed_real}, " \
+                             "owner: #{v[:owner]})")
+              return true
+            end
+          end
+
+          begin
+            not @node[:crowbar_wall][:claimed_disks][self.unique_name][:owner].to_s.empty?
+          rescue
+            false
+          end
```

Comment:
> Well, you have that in your new code, don't you? So it shouldn't matter.

Oh. Right. I played with that and forgot that I readded it. So the original code can be removed.

> But to answer your question: I think in the past, unclaiming was just making the owner empty. That's why the check is like this.

Hm. Sounds strange. So a Mash with claimed disks can have unclaimed disk. Great...